src/h5web/visualizations/heatmap/hooks.ts

```diff
@@ -58,12 +59,17 @@ export function useTextureData(): TextureDataState {
       return;
     }
 
+    // Keep existing texture data, if any
+    mergeState({ loading: true });
+
+    // Prepare transferable buffer to avoid cloning values array
+    const typedValues = Float64Array.from(values);
```

Comment:
Float 64 is the type for numbers in JS. Eventually, we may have to parse the JSON (or whatever format) returned by the server and create the appropriate typed array right away to optimise things further.